clients/src/test/java/org/apache/kafka/common/metrics/MetricsTest.java

```diff
@@ -308,12 +308,22 @@ public void testTimeWindowing() {
     @Test
     public void testOldDataHasNoEffect() {
```

Comment:
nit: i'd prefer a new testcase. this doesn't really have anything to do with 'testOldDataHasNoEffect'